src/PAK-Samples/MapReferenceToPaper/Program.cs

```diff
@@ -0,0 +1,415 @@
+                {
+                    if (b.Item1 > 0.8 && reducedChunk.Contains(b.Item4))
+                    {
+                        confidentMappings.Add(b);
+                        reducedChunk = reducedChunk.Replace(b.Item4, "");
+                    }
+                }
+
+                // For each mapping we're confident with, generate a mapping string with embedded XML tags for each match
+                foreach (var b in confidentMappings.OrderByDescending(a => a.Item1).ThenByDescending(a => a.Item4.Length))
+                {
+                    if (b.Item3 == b.Item4)
+                    {
+                        expandedChunk = Regex.Replace(expandedChunk, $"({Regex.Escape(b.Item4)})(?![^<]*>|[^<>]*</)", $" <attr confidence=\"{b.Item1}\" name=\"academic#{b.Item2}\">{b.Item4}</attr> ");
+                    }
+                    else
+                    {
+                        expandedChunk = Regex.Replace(expandedChunk, $"({Regex.Escape(b.Item4)})(?![^<]*>|[^<>]*</)", $" <attr confidence=\"{b.Item1}\" name=\"academic#{b.Item2}\" canonical=\"{b.Item3}\">{b.Item4}</attr> ");
+                    }
+                }
+
+                remainingUnmatched += (remainingUnmatched == "" ? "" : " ") + reducedChunk;
+                newParse = Regex.Replace(newParse, $"({Regex.Escape(unmatchedChunk)})(?![^<]*>|[^<>]*</)", expandedChunk);
+
+                // Collapse whitespace down to single space
+                remainingUnmatched = Regex.Replace(remainingUnmatched, "\\s+", " ").Trim();
+                newParse = Regex.Replace(newParse, "\\s+", " ").Trim();
+
+                var response = new ReferenceMapping
+                {
+                    PercentOfReferenceMapped = 1.0 - ((double)remainingUnmatched.Length / (double)interpretedQuery.Length),
+                    NormalizedReference = interpretedQuery,
+                    MappedPaper = entity,
+                    MappedReference = newParse,
+                    OriginalReference = referenceString,
+                    ReferenceNotMapped = remainingUnmatched
+                };
+
+                return response;
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Generates interpretation of query using Project Academic Knowledge Interpret method
+        /// </summary>
+        /// <param name="query">Natural language query to interpret</param>
+        /// <returns>JSON string of the Interpret method response. See https://docs.microsoft.com/en-us/academic-services/project-academic-knowledge/reference-interpret-method</returns>
+        static string GetQueryInterpretations(
+            string query,
+            string subscriptionKey,
+            bool complete = false,
+            int interpretationCount = 5,
+            int entityCount = 0,
+            string attributes = "",
+            int timeout = 500
+            )
+        {
+            var client = new HttpClient();
```

Comment:
>var client = new HttpClient(); [](start = 12, length = 30)

Can this be created once and shared?  If I remember correctly each instance of HttpClient holds onto a socket connection even after disposed for ~2mins.  .NET Core solved this with an HTTPClientFactory but I don't think it is available in .NET Framework.